x-pack/auditbeat/module/system/socket/_meta/data.json

```diff
@@ -0,0 +1,38 @@
+{
+    "@timestamp": "2017-10-12T08:05:34.853Z",
+    "agent": {
+        "hostname": "host.example.com",
+        "name": "host.example.com"
+    },
+    "destination": {
+        "ip": "10.0.2.15",
+        "port": 22
+    },
+    "event": {
+        "action": "existing_socket",
+        "dataset": "socket",
+        "id": "6aff69f8-7267-4604-9701-d7b67a7c65bc",
+        "kind": "state",
+        "module": "system"
+    },
+    "message": "Inbound socket (10.0.2.2:55270 -\u003e 10.0.2.15:22) OPEN by process sshd (PID: 22799) and user root (UID: 0)",
+    "network": {
+        "direction": "inbound",
+        "type": "ipv4"
+    },
+    "process": {
+        "name": "sshd",
+        "pid": 22799
+    },
+    "service": {
+        "type": "system"
+    },
+    "source": {
```

Comment:
In Filebeat we had to use `source_ecs` instead of `source`, because there was a hard conflict. It used to be a string field and in 7.x/ECS becomes an object field.

Auditbeat doesn't use `source` in 6.x, so we could simply introduce it right now and be done with this. It affects only the Auditbeat index, so it may be fine. If we take this route, querying on a field in `source` will mean the user needs to query both `source` for Auditbeat and `source_ecs` for Filebeat.

So even if there's no actual conflict, there is an advantage in prioritizing consistency instead, and naming this field `source_ecs` in 6.x for Auditbeat, then migrating it for 7.x.

WDYT @ruflin @tsg ?

Personally I think it may be fine to introduce `source` here and not have to do a transition of this new field right away for 7.0. I don't think using Auditbeat and Filebeat will be a big enough use case to warrant this in the short period we will have this inconsistency. SecOps will only work on 7.x anyway.

As far as I can tell, this PR does not introduce the fields definition for `source.*`, so if we decide to go this direction, @cwurm you should ensure to add them, but only in the context of Auditbeat.

Reply:
I'm ok with introducing `source.*` in auditbeat only. If we do it definitively should follow ECS.

Reply:
Auditbeat is [already using](https://www.elastic.co/guide/en/beats/auditbeat/current/exported-fields-auditd.html#_source_fields) `source` in 6.x, so we should be good. But there are a few other fields that I'll add in a commit momentarily.

Reply:
@cwurm Gotcha.

These field definitions are not quite in line with [ECS' source field set](https://github.com/elastic/ecs#source), however. There's no incompatibility between the two, but we're also missing field definitions for source.